azure-client-runtime/src/main/java/com/microsoft/azure/v2/AzureProxy.java

```diff
@@ -161,21 +172,13 @@ else if (returnTypeToken.isSubtypeOf(Observable.class)) {
                                             .takeUntil(new Func1<OperationStatus<Object>, Boolean>() {
                                                 @Override
                                                 public Boolean call(OperationStatus<Object> operationStatus) {
-                                                    final boolean stop = operationStatus.isDone();
-                                                    if (stop) {
-                                                        // Take until will not return the operationStatus that is
-                                                        // marked as done, so we set it here and then concatWith() it
-                                                        // later to force the Observable to return the operationStatus
-                                                        // that is done.
-                                                        lastOperationStatus.set(operationStatus);
-                                                    }
-                                                    return stop;
+                                                    return operationStatus.isDone();
                                                 }
                                             })
                                             .concatWith(Observable.defer(new Func0<Observable<OperationStatus<Object>>>() {
```

Comment:
Interesting.. as per doc [takeUntil](http://reactivex.io/documentation/operators/images/takeUntil.p.png) is suppose to forward the event meeting the criteria as final event

```java
final int[] i = { 0 };
Observable.defer(() -> Observable.just(i[0]++))
            .repeat()
            .takeUntil(v -> v == 5)
            .subscribe(System.out::println);
```

output is

```
0
1
2
3
4
5
```

Anyway, if we find that concat is needed can we add some comments (seems we already had some note but it got deleted in this pr)?

Reply:
That's bizarre. I just removed the concatWith() call and all of my tests pass. I remember needing to put that in, but I must've been using a different operator and switched it to takeUntil(). Weird. I've removed the concatWith() now.